crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -399,6 +399,9 @@ async fn c8y_mapper_syncs_pending_alarms_on_startup() {
     )
     .await;
 
+    // wait for messages to get synced properly before aborting mapper
+    tokio::time::sleep(Duration::from_millis(300)).await;
```

Comment:
Instead of this hard wait, it'd be better to wait till the mapper sends the copy of the `temperature_alarm` to `c8y-internal/alarms/critical/temperature_alarm` topic as well. That's the final confirmation that the mapper has received the temperature alarm and processed it.

Reply:
Added a subscriber to wait on the messages to sync on the `c8y-internal/alarms/critical/temperature_alarm` topic before stopping the broker. But still, this can not wait forever we still have to put some timeout,